backend/src/Users/users.controller.ts

```diff
@@ -0,0 +1,40 @@
+import { Body, Controller, Get, Logger, Param, Post } from '@nestjs/common';
+import { UsersService } from './users.service';
+import {
+  FriendRequestDto,
+  UserCreateReqDto,
+  UserCreateResDto,
+  UserGetDto,
+} from 'src/common/dto/users.dto';
+
+@Controller('user')
+export class UsersController {
+  constructor(private service: UsersService) {}
+
+  @Post()
+  createUser(@Body() body: UserCreateReqDto): Promise<UserCreateResDto> {
+    return this.service.create(body);
+  }
+
+  @Get(':id')
+  getUserById(@Param('id') param: string): Promise<UserGetDto> {
```

Comment:
validateParamId関数を自作しなくても、controller側にPipeを書くことでうまくやってくれるみたいです！
https://docs.nestjs.com/pipes#built-in-pipes

```suggestion
  getUserById(@Param('id', ParseIntPipe) param: string): Promise<UserGetDto> {
```

（僕も今調べて知ったので、色々反映させないと・・・）

Reply:
もしかするとオートバリデーションっていうの使った方が楽かもです。
https://zenn.dev/kisihara_c/books/nest-officialdoc-jp/viewer/techniques-validation#%E3%82%AA%E3%83%BC%E3%83%88%E3%83%90%E3%83%AA%E3%83%87%E3%83%BC%E3%82%B7%E3%83%A7%E3%83%B3

Reply:
ありがとうございます。追加します！